syntax = "proto3";

package XXMsg;

import "TestMsgDefine.proto";

message TestCommonInfo
{
    bool   test_bool   = 1;  // C++中的bool     -> 布尔值
    float  test_float  = 2;  // C++中的float    -> 单精度浮点数
    double test_double = 3;  // C++中的double   -> 双精度浮点数
    bytes  test_string = 4;  // C++中的string   -> 可能包含任意顺序的字节数据
    int32  test_int32  = 5;  // C++中的int32_t  -> 使用可变长编码，编码负数时不够高效，如果多用于负数则使用sint32
    sint32 test_sint32 = 6;  // C++中的int32_t  -> 使用ZigZag和可变长编码，编码负数时比int32高效
    uint32 test_uint32 = 7;  // C++中的uint32_t -> 无符号类型，使用可变长编码
    int64  test_int64  = 8;  // C++中的int64_t  -> 使用可变长编码，编码负数时不够高效，如果多用于负数则使用sint64
    sint64 test_sint64 = 9;  // C++中的int64_t  -> 使用ZigZag和可变长编码，编码负数时比int64高效
    uint64 test_uint64 = 10; // C++中的uint64_t -> 无符号类型，使用可变长编码
}

message TestCommonInfoList
{
    repeated TestCommonInfo common_info_list = 1;
    repeated TestCommonType common_type_list = 2;
}